zenip-42202.md

```diff
@@ -0,0 +1,119 @@
+
+- The `epoch` values of certificates must be consecutive (i.e. after a certificate of epoch N the sidechain must publish a certificate of epoch N + 1)
+
+- The `quality` mechanism is disabled, so it’s not possible to publish two or more certificates for the same epoch
+
+These non-ceasable sidechains must satisfy the following constraints:
+
+- **Sidechain Creation**
+
+  - `withdrawalEpochLength` must be set to 0 (to disable the fixed-length epochs and submission windows, otherwise they work exactly as sidechains v1 do).
+
+  - `mainchainBackwardTransferRequestDataLength` must be set to 0 (to disable the MBTR).
+
+  - `wCeasedVk` must be empty (to disable the CSW).
+
+- **Certificates**
+
+  - The `quality` field must always be set to “0”.
+
+  - If mainchain receives first a certificate of epoch N (either in mempool or in a block) and then another certificate of same epoch, the latter is rejected.
+
+  - All certificates must refer (through the `endEpochCumScTxCommTreeRoot` field) to different ordered block. So, for instance, if the mainchain receives a certificate for epoch N referring to the mainchain block M (i.e. setting the `endEpochCumScTxCommTreeRoot `field to the same value specified in the header of the mainchain block M), any certificate coming next must refer to a `endEpochCumScTxCommTreeRoot` (and thus to a block) with higher height than M (M + 1, M + 2, etc., are all good candidates). Such constraint enforces that a certificate of higher epoch refers to a more updated state of the mainchain and of the sidechain.
+
+  - More than one certificate for the same sidechain can be included in the same block (if they respect all the constraints above).
+
+When sidechains v2 are created with `withdrawalEpochLength` not set to 0, they will enable the default ceasing mechanism as it is intended for sidechains v1.
+
+### Maturity
+
+For sidechain v0 and v1, Zendoo implements a maturity mechanism for the amount of coins transferred to and from any sidechain.
+
+For instance, whenever transactions are created with a cross-chain output (sidechain creation, forward transfer, or mainchain backward transfer request), the coins and fees are able to move to the sidechain but the amount is not immediately added to its balance.  The coins are marked as immature and after some block confirmation, they are included in the sidechain balance. The number of confirmations is set to `10` blocks for mainnet in the chain parameters.
+
+The same concept applies to backward transfers in a certificate. During the submission window phase, a certificate can be superseded by another certificate of higher quality. In this case, it makes sense to mark the backward transfers as immature until the certificate is final.
+
+Moreover, a certificate can be reverted back, just in case a sidechain ceases. Zendoo considers maturity of any backward transfer included in a top quality certificate for epoch N only after the end of the submission window for epoch N + 1.
```

Comment:
```suggestion
In case a sidechain ceases any backward transfers in an immature certificate are reverted. Zendoo only considers backward transfers included in a top quality certificate for epoch N mature at the end of the submission window for epoch N + 1 (another certificate was submitted in submission window of epoch N + 1).
```

Reply:
If a sidechain ceases, any backward transfers in an immature certificate are reverted. Zendoo only considers backward transfers included in a top quality certificate for epoch N mature at the end of the submission window for epoch N + 1 (submitted another certificate in submission window of epoch N + 1).

Reply:
Thanks @Ramone-HL, I've applied your suggestion, I only expanded a bit the text between brackets:

`(once a new certificate is submitted during the submission window of epoch N + 1)`.